### PR TITLE
42 ec user cannot request own

### LINKED ISSUE #42: ec user cannot request own
```
As a user 
In order to not get confused
I would like to not be able to book my own room
```

### PR DESCRIPTION
**Pull request by tobenna**
closes #42

### CODE REVIEW

File: spec/features/approve_booking_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    request.save
+  end
+
+  scenario 'can see all requests for spaces' do
+    visit '/requests'
+    expect(page).to have_content('mySpace')
+    expect(page).to_not have_content('House 2')
+  end
+
+  scenario 'can only see my space requests' do
```
Comment (kevinpmcc) [inline spec/features/approve_booking_spec.rb]: This is fine for this user story but we'll need to amend/ delete this test when we implement 'view requests I've made'

Comment (tobenna) [inline spec/features/approve_booking_spec.rb]: :100: makes I see what you mean. It would break the test when we start separating the types of requests

Comment (kevinpmcc) [inline spec/features/approve_booking_spec.rb]: We can fix it in that user story view requests I've made.  Spent a little time o  it last night. Think I know what we need

File: app/models/datamapper_setup.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# DataMapper::Logger.new($stdout, :debug)
```
Comment (olmesm) [inline app/models/datamapper_setup.rb]: can we remove this?

File: app/controllers/requests.rb [hunk 3]
```diff
@@ -1,10 +1,33 @@
...
+      status 404
+    else
+      req.save
+      redirect('/spaces')
+    end
+  end
+
+  get '/requests' do
+    # @BookingRequests_received = Space.all(user: current_user)
+    @requests = User.get(session[:user_id]).spaces.booking_requests
```
Comment (olmesm) [inline app/controllers/requests.rb]: nice use of the code I wrote for you :+1:

Comment (tobenna) [inline app/controllers/requests.rb]: I know! you inspire me

**Pull request merged**